boa_engine/src/object/mod.rs

```diff
@@ -1839,7 +1849,7 @@ impl Object {
 
     /// Helper function for property removal.
     #[inline]
-    pub(crate) fn remove(&mut self, key: &PropertyKey) -> Option<PropertyDescriptor> {
+    pub(crate) fn remove(&mut self, key: &PropertyKey) -> bool {
```

Comment:
Here we should add a documentation comment similar to the one in `insert()`, explaining what it returns too.